src/extra/components/adq.py

```diff
@@ -356,23 +356,31 @@ def _preprocess(self, data, out):
 
         return out
 
-    def _prepare_pulses(self):
+    def _prepare_pulses(self, train_ids):
```

Comment:
Why not have train_ids by default None and revert to the old behaviour to make it backwards compatible?

Reply:
This is a private method, and I've fixed all its call sites.